webapp/shop/decorators.py

```diff
@@ -117,7 +118,7 @@ def decorated_function(*args, **kwargs):
                     return flask.redirect(f"/login?next={redirect_path}")
 
             ua_contracts_api = get_ua_contracts_api_instance(
-                user_token, response, session
+                user_token, response, session, flask.request.remote_addr
```

Comment:
You may want to look at a different field instead: https://stackoverflow.com/questions/12770950/flask-request-remote-addr-is-wrong-on-webfaction-and-not-showing-real-user-ip

But it will all depend on how the content-cache/K8s ingress will be configured in front of it.